tests/bluetooth/tester/src/gap.c

```diff
@@ -184,6 +188,98 @@ static void controller_info(uint8_t *data, uint16_t len)
 		    CONTROLLER_INDEX, (uint8_t *) &rp, sizeof(rp));
 }
 
+#if !defined(CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY)
+static const char *oob_config_str(int oob_config)
+{
+	switch (oob_config) {
+	case BT_CONN_OOB_LOCAL_ONLY:
+		return "Local";
+	case BT_CONN_OOB_REMOTE_ONLY:
+		return "Remote";
+	case BT_CONN_OOB_BOTH_PEERS:
+		return "Local and Remote";
+	case BT_CONN_OOB_NO_DATA:
+	default:
+		return "no";
+	}
+}
+#endif /* !defined(CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY) */
+
+static void oob_data_request(struct bt_conn *conn,
+			     struct bt_conn_oob_info *oob_info)
+{
+	struct bt_conn_info info;
+	int err = bt_conn_get_info(conn, &info);
+	if (err) {
+		return;
+	}
+
+	char addr[BT_ADDR_LE_STR_LEN];
+
+#if !defined(CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY)
+	if (oob_info->type == BT_CONN_OOB_LE_SC) {
+		struct bt_le_oob_sc_data *oobd_local =
+			oob_info->lesc.oob_config != BT_CONN_OOB_REMOTE_ONLY ?
+				      &oob_local.le_sc_data :
+				      NULL;
+		struct bt_le_oob_sc_data *oobd_remote =
+			oob_info->lesc.oob_config != BT_CONN_OOB_LOCAL_ONLY ?
+				      &oob_remote.le_sc_data :
+				      NULL;
+
+		if (oobd_remote) {
+			/* TODO: This is a temp hack,
+			 * the address should be part of gap_oob_sc_set_remote_data_cmd
+			 */
+			bt_addr_le_copy(&oob_remote.addr, info.le.remote);
+		}
+
+		if (oobd_local &&
+		    bt_addr_le_cmp(info.le.local, &oob_local.addr)) {
+			bt_addr_le_to_str(info.le.local, addr, sizeof(addr));
+			LOG_DBG("No OOB data available for local %s",
+				log_strdup(addr));
+			bt_conn_auth_cancel(conn);
+			return;
+		}
+
+		bt_le_oob_set_sc_data(conn, oobd_local, oobd_remote);
+		bt_addr_le_to_str(info.le.dst, addr, sizeof(addr));
+		LOG_DBG("Set %s OOB SC data for %s, ",
+			oob_config_str(oob_info->lesc.oob_config),
+			log_strdup(addr));
+
+		return;
+	}
+#endif /* CONFIG_BT_SMP_OOB_LEGACY_PAIR_ONLY */
+
+	bt_addr_le_to_str(info.le.dst, addr, sizeof(addr));
+	LOG_DBG("Legacy OOB TK requested from remote %s", addr);
+}
+
+static void get_oob_sc_local_data(void)
+{
+	struct gap_oob_sc_get_local_data_rp rp = { 0 };
+	memcpy(&rp.conf, oob_local.le_sc_data.c, sizeof(rp.conf));
```

Comment:
Keep the same style as set_oob_sc_remote_data?
memcpy(&rp.conf[0], &oob_local.le_sc_data.c[0], sizeof(rp.conf));

Reply:
will do